src/utils/oppfolgingsplanerUtils.js

```diff
@@ -0,0 +1,53 @@
+const oppfolgingsplanerValidNow = (oppfolgingsplaner) => {
+    return oppfolgingsplaner.filter((plan) => {
+        return new Date(plan.godkjentPlan.gyldighetstidspunkt.tom) > new Date() && plan.godkjentPlan.deltMedNAV;
+    });
+};
+
+const planForVirksomhetsnummer = (oppfolgingsplaner, virksomhetsnummer) => {
+    return oppfolgingsplaner.find((plan) => {
+        return plan.virksomhet.virksomhetsnummer === virksomhetsnummer;
+    });
+};
+
+const removePlanFromList = (oppfolgingsplaner, id) => {
+    return oppfolgingsplaner.filter((plan) => {
+        return plan.id !== id;
+    });
+};
+
+const replacePlanInListIfCandidateIsNewer = (oppfolgingsplaner, existingPlan, candidatePlan) => {
+    let planerListWithoutEarlierPlanFromVirksomhet = oppfolgingsplaner;
+    if (new Date(candidatePlan.godkjentPlan.deltMedNAVTidspunkt) > new Date(existingPlan.godkjentPlan.deltMedNAVTidspunkt)) {
+        planerListWithoutEarlierPlanFromVirksomhet = removePlanFromList(oppfolgingsplaner, existingPlan.id);
+        planerListWithoutEarlierPlanFromVirksomhet.push(candidatePlan);
+    }
+
+    return planerListWithoutEarlierPlanFromVirksomhet;
+};
+
+const putNewestPlanInList = (oppfolgingsplaner, currentlyNewestPlanForVirksomhet, candidatePlan) => {
+    if (!currentlyNewestPlanForVirksomhet) {
+        oppfolgingsplaner.push(candidatePlan);
+        return oppfolgingsplaner;
+    }
+
+    return replacePlanInListIfCandidateIsNewer(oppfolgingsplaner, currentlyNewestPlanForVirksomhet, candidatePlan);
+};
+
+const newestPlanForEachVirksomhet = (oppfolgingsplaner) => {
+    let newestPlanPerVirksomhet = [];
+
+    oppfolgingsplaner.forEach((candidatePlan) => {
+        const currentlyNewestPlanForVirksomhet = planForVirksomhetsnummer(newestPlanPerVirksomhet, candidatePlan.virksomhet.virksomhetsnummer);
+
+        newestPlanPerVirksomhet = putNewestPlanInList(newestPlanPerVirksomhet, currentlyNewestPlanForVirksomhet, candidatePlan);
+    });
+
+    return newestPlanPerVirksomhet;
+};
+
+export const activeOppfolgingsplaner = (oppfolgingsplaner) => {
+    const newestPlans = newestPlanForEachVirksomhet(oppfolgingsplaner);
```

Comment:
Er det mulig å heller kjøre en sort på hele listen og så ta en  "find first" for hvert virksomhetsnummer?

Reply:
Ja, det høres bra ut 👍